pkg/pulp_client/task.go

```diff
@@ -0,0 +1,128 @@
+package pulp_client
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"regexp"
+	"time"
+
+	zest "github.com/content-services/zest/release/v3"
+)
+
+const (
+	COMPLETED string = "completed"
+	WAITING   string = "waiting"
+	RUNNING   string = "running"
+	SKIPPED   string = "skipped"
+	CANCELED  string = "canceled"
+	FAILED    string = "failed"
+)
+
+// GetTask Fetch a pulp task
+func (r pulpDaoImpl) GetTask(taskHref string) (zest.TaskResponse, error) {
+	task, _, err := r.client.TasksApi.TasksRead(r.ctx, taskHref).Execute()
+
+	if err != nil {
+		return zest.TaskResponse{}, err
+	}
```

Comment:
I haven't checked zest, but does this or anything similar block while waiting for a response? And if so would it timeout if it takes too long? Either here or on the zest side of things.